packages/website/blog/2023-03-13-announcing-typescript-eslint-v6-beta.md

```diff
@@ -0,0 +1,309 @@
+- ⏳ [feat(eslint-plugin): [prefer-optional-chain] use type checking for strict falsiness](https://github.com/typescript-eslint/typescript-eslint/pull/6240): Fixes edge case bugs in the [`@typescript-eslint/prefer-optional-chain`](https://typescript-eslint.io/prefer-optional-chain) rule around false positives, at the cost of making it require type information.
+
+### Tooling Breaking Changes
+
+- [feat(typescript-estree): deprecate createDefaultProgram](https://github.com/typescript-eslint/typescript-eslint/pull/5890): Renames `createDefaultProgram` to `deprecated__createDefaultProgram`, with associated `@deprecated` TSDoc tags and warnings.
+- [feat: drop support for node v12](https://github.com/typescript-eslint/typescript-eslint/pull/5918)
+- [feat: bump minimum supported TS version to 4.2.4](https://github.com/typescript-eslint/typescript-eslint/pull/5915): this matches [DefinitelyTyped's 2-year support window](https://github.com/DefinitelyTyped/DefinitelyTyped#support-window).
+- [chore: drop support for ESLint v6](https://github.com/typescript-eslint/typescript-eslint/pull/5972)
+
+## Developer-Facing Changes
+
+typescript-eslint v6 comes with a suite of cleanups and improvements for developers as well.
+If you author any ESLint plugins or other tools that interact with TypeScript syntax, then we recommend you try out typescript-eslint v6 soon.
+It includes some breaking changes that you may need to accommodate for.
+
+:::tip
+If you're having trouble working with the changes, please let us know on [the typescript-eslint Discord's `v6` channel](https://discord.com/channels/1026804805894672454/1084245444676292688)!
+:::
+
+### Type Checker Wrapper APIs
+
+As described in our [ASTs and typescript-eslint](/blog/asts-and-typescript-eslint) post, ESLint rules don't natively work with AST nodes compatible with TypeScript's API.
+Retrieving type information for an ESLint AST node in a custom rule requires code somewhat like:
+
+```ts title="custom-rule-with-v5.ts"
+{
+  // ...
+  create() {
+    const services = util.getParserServices(context);
+    const checker = services.program.getTypeChecker();
+    const tsNode = services.esTreeNodeToTSNodeMap.get(esNode);
+    const type = checker.getTypeAtLocation(node);
+
+    // ...
+  }
+  // ...
+}
+```
+
+How cumbersome, just to call to a single method (`getTypeAtLocation`) on the TypeScript API!
+
+In typescript-eslint v6, we've added a set of wrapper APIs on the `services: ParserServices` object that act as shortcuts for commonly used TypeScript APIs including `getTypeAtLocation`:
+
+```ts title="custom-rule-with-v6.ts"
+{
+  // ...
+  create() {
+    const services = util.getParserServices(context);
+    const type = services.getTypeAtLocation(node);
+
+    // ...
+  }
+  // ...
+}
+```
+
+For now, the available wrapper APIs are:
+
+- `getSymbolAtLocation`: directly passes an ESTree node to TypeScript's `checker.getSymbolAtLocation`
```

Comment:
nit: the language here is a little confusing - it's _indirect_ right? It gets converted to its tsnode equivalent first before being passed to TypeScript's checker method